pandas/core/arrays/categorical.py

```diff
@@ -2167,8 +2167,7 @@ def _reverse_indexer(self):
         r, counts = libalgos.groupsort_indexer(self.codes.astype('int64'),
                                                categories.size)
         counts = counts.cumsum()
-        result = [r[counts[indexer]:counts[indexer + 1]]
-                  for indexer in range(len(counts) - 1)]
+        result = (r[start:end] for start, end in zip(counts, counts[1:]))
```

Comment:
Should the first zip arg be counts[:-1]?

Reply:
`zip` stops when the shortest sequence is exhausted, so as written should be equivalent to if `counts[:-1]` was the first argument instead:
```python
In [1]: list(zip('abcd', 'abc'))
Out[1]: [('a', 'a'), ('b', 'b'), ('c', 'c')]
```
Not against changing if the consensus is that `counts[:-1]` is more readable though.

Reply:
Doesn't really matter. Since the 2nd iterator is smaller than the first, `zip` will be exhausted once the second iterator is done.

Reply:
Thanks for the reminder.  I must still be used to the py2 rules